com.arc.embeddedcdt/src/com/arc/embeddedcdt/common/DebuggerPluginVersionsChecker.java

```diff
@@ -0,0 +1,61 @@
+package com.arc.embeddedcdt.common;
+
+import java.util.HashSet;
+
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Display;
+
+import com.arc.embeddedcdt.LaunchConfigurationConstants;
+import com.arc.embeddedcdt.dsf.utils.Configuration;
+
+public class DebuggerPluginVersionsChecker {
+
+    private HashSet<String> seenLaunchConfigurationNames = new HashSet<>();
+
+    private static DebuggerPluginVersionsChecker INSTANCE = null;
+
+    private ILaunchConfiguration launchConfiguration;
+
+    private DebuggerPluginVersionsChecker() {
+    }
+
+    public static DebuggerPluginVersionsChecker getDebuggerPluginVersionsChecker() {
```

Comment:
IMHO, can be just `getInstance` for singletons - name of an instance type is already in the class name. Such long name would make sense if instance was created by some different class, like factory or facade, but here it looks like too much.